### PR TITLE
Updated .NET documentation

### PR DESCRIPTION
**Pull request by cormacpayne**
- Updated README with information about repo structure and (private) links to on-boarding process

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@cormacpayne__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

**Comment by markcowl** [comment]
@azuresdkci add to whitelist

File: README.md [hunk 1]
```diff
@@ -43,17 +40,49 @@ Using the command line:
...
-  - In your library csproject file, set the msbuild property "AutoRestProjects" to "true"
-  - In your test project files, set both "AutoRestProjects" and "SDKTestProject" to "true"
-  - To simplify test discovery, the test folder must be named with ".tests"
+## To on-board new libraries
+
+### Project Structure
+
+In "src\ResourceManagement", you will find packages for services that have already been implemented
+
+  - Each service contains a package for their generated code, as well as a package for tests
```
Comment (markcowl) [inline README.md]: project or folder, rather than package, I think

File: README.md [hunk 2]
```diff
@@ -43,17 +40,49 @@ Using the command line:
...
+
+The **AutoRest** branch contains the code generated from AutoRest.
+
+The **master** branch contains the code generated from Hydra/Hyak.
+  - Hydra/Hyak is Azure's legacy code generation technology.
+  - This can still be used to generate client libraries, but the project is not being advanced in favor of AutoRest. Your team should move to AutoRest and Swagger as soon as possible.
+
+### Standard Process
+
+For a step-by-step process on how to generate C# code for your service and get it published, click [here](https://github.com/Azure/adx-documentation-pr/blob/master/engineering/autorest-to-powershell.md#phase-1---swagger-specs-to-net-sdk-using-autorest).
```
Comment (markcowl) [inline README.md]: Note:  External contributors are not going to have or be able to get access to this repo.  We should think about moving contributions guidelines / process into this repo

File: README.md [hunk 3]
```diff
@@ -43,17 +40,49 @@ Using the command line:
...
+
+**Note**: if you receive a 404 error trying to access this page, click [here](https://aka.ms/azuregithub) and sign in with your Microsoft corporate account to gain access.
+
+### Code Review Process
+
+To see the requirements and process for checking code into the working branch, click [here](https://github.com/Azure/adx-documentation-pr/blob/master/engineering/adx_sdk_gitflow.md#checking-code-into-the-dev-working-branch)
```
Comment (markcowl) [inline README.md]: Same comment

File: README.md [hunk 4]
```diff
@@ -43,17 +40,49 @@ Using the command line:
...
+
+**Note**: if you receive a 404 error trying to access this page, click [here](https://aka.ms/azuregithub) and sign in with your Microsoft corporate account to gain access.
+
+### Adding Tests
+
+Regarding the test project, one thing that's important is to name the test project folder and related project assembly to be ".tests".
```
Comment (markcowl) [inline README.md]: Name the test project by adding a ".Tests" suffix to the product assembly name.  For example, the test project for 'Microsoft.Azure.Management.Compute' should be named 'Microsoft.Azure.Management.Compute.Tests'

File: README.md [hunk 5]
```diff
@@ -43,17 +40,49 @@ Using the command line:
...
+
+  - This is for improving CI performance so to find exactly one copy of your test assembly.
+  - Also, due to test dependencies, the test project should build .NET 4.5. For example, check out "src\ResourceManagement\Resource\Resource.tests"
```
Comment (markcowl) [inline README.md]: The tests project should build both .net4.5 and netstandard1.5

File: README.md [hunk 6]
```diff
@@ -43,17 +40,49 @@ Using the command line:
...
 
 ### Issues with Generated Code
-Much of the SDK code is generated from metadata specs about the REST APIs. Do not submit PRs that modify generated code. Instead, file an issue describing the problem, OR refer to the the [AutoRest project](AutoRest) to view and modify the generator. 
 
-[AutoRest]:https://github.com/azure/autorest
+Much of the SDK code is generated from metadata specs about the REST APIs. Do not submit PRs that modify generated code. Instead, file an issue describing the problem, OR refer to the the [AutoRest project](https://github.com/azure/autorest) to view and modify the generator.
```
Comment (markcowl) [inline README.md]: Note that generated code uses partial classes, so that Customizations can be added to enhance the generated code.  You can add additional methods, properties, and overloads to the SDK by adding classes in the 'Customizations' folder of a project.

File: README.md [hunk 7]
```diff
@@ -43,17 +40,49 @@ Using the command line:
...
-  - To simplify test discovery, the test folder must be named with ".tests"
+## To on-board new libraries
+
+### Project Structure
+
+In "src\ResourceManagement", you will find packages for services that have already been implemented
+
+  - Each service contains a package for their generated code, as well as a package for tests
+  - The file 'generate.cmd', used to generate library code for the given service, can also be found in these packages
+
```
Comment (markcowl) [inline README.md]: The Folder 'Generated' contains generated code, The Folder 'Customizations' contains additions to the generated code - this can include additions to the genearetd partial classes, or additional classes that augment the SDK or call generated code.

**Comment by markcowl** [comment]
@cormacpayne A few comments

**Comment by markcowl** [comment]
@azuresdkci retest this please

**Comment by cormacpayne** [comment]
@markcowl resolved your review comments

File: README.md [hunk 8]
```diff
@@ -43,17 +40,94 @@ Using the command line:
...
+ 3. Create your Swagger specification for your HTTP API. For more information see 
+ [Introduction to Swagger - The World's Most Popular Framework for APIs](http://swagger.io)
+ 4. Install the latest version of AutoRest and use it to generate your C# client. For more info on getting started with AutoRest, 
+ see the [AutoRest repository](https://github.com/Azure/autorest)
+ 5. Create a branch in your fork of Azure SDK for .NET and add your newly generated code to your project. If you don't have
+ a project in the SDK yet, look at some of the existing projects and build one like the others. 
+ 6. **MANDATORY**: Add or update tests for the newly generated code.
+ 7. Once added to the Azure SDK for .NET, Build your local package using command "msbuild build.proj /t:build;package /p:scope=YourService" 
+ (Note, 'YourService' comes from the sub folder under <sdk-repo-root>\src, for example: "ResourceManagement\Compute")
+ 8. Bump up the package version in YourService.nuget.proj
```
Comment (hovsepm) [inline README.md]: >Bump up the package version in YourService.nuget.proj [](start = 4, length = 53)

This is true only for master branch. For Autorest branch they will need to change version in project.json file and also AssemblyInfo.cs file.

File: README.md [hunk 9]
```diff
@@ -43,17 +40,94 @@ Using the command line:
...
+
+- Prior to issuing the pull request:
+  - All code must have completed any necessary legal signoff for being publically viewable (Patent review, JSR review, etc.)
+  - The changes cannot break any existing functional /unit tests that are part of the central repository.
+    - This includes all tests, even those not associated with the given feature area.
+  - Code submitted must have basic unit test coverage, and have all the unit tests pass. Testing is the full responsibility of the service team
+    - Functional tests are encouraged, and provide teams with a way to mitigate regressions caused by other code contributions.
+  - Code should be commented.
+  - Code should be fully code reviewed.
+  - Code should be able to merge into the dev branch being targeted.
```
Comment (hovsepm) [inline README.md]: >Code should be able to merge into the dev branch being targeted. [](start = 4, length = 64)

Please rephrase this to indicate that we want partners to have a PR without merge conflicts.

**Comment by hovsepm** [comment]
> 

Delete this secret information. 

---
Refers to: README.md:134 in a638692. [](commit_id = a6386928a54d7814919e75261a9ff739e34200f5, deletion_comment = False)

**Comment by markcowl** [comment]
LGTM once Hovesp's comments are addressed.
